pkg/reconciler/subscription/subscription.go

```diff
@@ -293,50 +312,38 @@ func (r *reconciler) syncPhysicalChannel(ctx context.Context, sub *v1alpha1.Subs
 	return nil
 }
 
-func (r *reconciler) listAllSubscriptionsWithPhysicalChannel(ctx context.Context, sub *v1alpha1.Subscription) ([]v1alpha1.Subscription, error) {
+func (r *Reconciler) listAllSubscriptionsWithPhysicalChannel(ctx context.Context, sub *v1alpha1.Subscription) ([]v1alpha1.Subscription, error) {
 	subs := make([]v1alpha1.Subscription, 0)
 
-	opts := &client.ListOptions{
-		Namespace: sub.Namespace,
-		// Set Raw because if we need to get more than one page, then we will put the continue token
-		// into opts.Raw.Continue.
-		Raw: &metav1.ListOptions{},
-	}
-	for {
-		sl := &v1alpha1.SubscriptionList{}
-		err := r.client.List(ctx, opts, sl)
-		if err != nil {
-			return nil, err
-		}
-		for _, s := range sl.Items {
-			if sub.UID == s.UID {
-				// This is the sub that is being reconciled. Skip it.
-				continue
-			}
-			if equality.Semantic.DeepEqual(sub.Spec.Channel, s.Spec.Channel) {
-				subs = append(subs, s)
-			}
+	sl, err := r.subscriptionLister.Subscriptions(sub.Namespace).List(labels.Everything()) // TODO: we can use labels to help here
```

Comment:
For future reference, the continue token handling can be removed based on https://github.com/knative/eventing/pull/1044#discussion_r275572557.